scrapy_fake_useragent/middleware.py

```diff
@@ -1,77 +1,89 @@
 import logging
-from fake_useragent import UserAgent
 
 from scrapy.downloadermiddlewares.retry import RetryMiddleware
 from scrapy.utils.response import response_status_message
+from scrapy.utils.misc import load_object
 
 logger = logging.getLogger(__name__)
 
-
-class RandomUserAgentMiddleware(object):
+class RandomUserAgentBase:
     def __init__(self, crawler):
-        super(RandomUserAgentMiddleware, self).__init__()
+        self._ua_provider = self._get_provider(crawler)
+        self._per_proxy = crawler.settings.get('RANDOM_UA_PER_PROXY', False)
+        self._proxy2ua = {}
+
+    def _get_provider(self, crawler):
+        self.providers_paths = crawler.setting.get('FAKEUSERAGENT_PROVIDERS', None)
+
+        assert((self.providers_paths is not None) and len(self.providers_paths) > 0,
```

Comment:
For backwards compatibility, we need to make sure we have a default value for `FAKEUSERAGENT_PROVIDERS` set to be just single `FakeUserAgent` provider. So that users who receive our package update don't experience issues but also have an ability to work it around with `FAKEUSERAGENT_PROVIDERS` and a set of new built-in providers or with a custom provider.